### ISSUE TITLE
Sorting tokens in fixture does not work

### ISSUE THREAD

**Issue by sz-piotr** [Uniswap/uniswap-v2-core#99]
https://github.com/Uniswap/uniswap-v2-core/blob/4dd59067c76dea4a0e8e4bfdda41877a6b16dedc/test/shared/fixtures.ts#L40

Because ethers returns a string from this method call, `.address` always results in undefined, thus the order of tokens is never changed.

**Comment by moodysalem** [comment]
Good catch! I guess token0 and token1 must always be the same for the fixture, so we never had failing tests out of it.

**Comment by Devilla** [comment]
> Good catch! I guess token0 and token1 must always be the same for the fixture, so we never had failing tests out of it.

Sir, how can token0 and token1 be same for pairFixture ?

**Comment by moodysalem** [comment]
I mean same addresses across test runs (but token0 != token1), since we use the same mnemonic for every test run

**Comment by Devilla** [comment]
Okay so it'll fail for different pair address and pass for same pair address right?

**Comment by ghost** [comment]
I think the best fix would be to lowercase the addresses of the token and just compare like how it is done from the factory contract as well.

**Comment by brkbrk700** [comment]
https://github.com/Uniswap/uniswap-v2-core

**Comment by brkbrk700** [comment]
https://github.com/Uniswap/uniswap-v2-core

[package.json](https://github.com/user-attachments/files/23106371/package.json)
